### PR TITLE
docs(@angular/cli): add cherry-pick instructions.

### PR DESCRIPTION
**Pull request by dgp1130**
This includes instructions on how to cherry-pick commits into the patch branch. Since `cherry-pick` isn't that commonly used, it's useful to write this down for developers who might not be that familiar with it. It also includes fetching the commit beforehand, so users don't get "bad object" errors which can be annoying to work around.

The `-x` option is used as well to include a reference to the commit the cherry-pick came from.

### CODE REVIEW

File: docs/process/release.md [hunk 1]
```diff
@@ -39,6 +39,27 @@ Say the following PR is merged;
 
 Only the first 2 commits should be cherry picked to the patch branch, as the last 3 are related to a new feature.
 
+Cherry picking is done by checking out the patch branch and cherry picking the new commit onto it.
+This should be done after merging to master.
+
+```shell
+# Make sure commit to upstream/master is present local repo.
```
Comment (kyliau) [inline docs/process/release.md]: is present **in** local repo

File: docs/process/release.md [hunk 2]
```diff
@@ -39,6 +39,27 @@ Say the following PR is merged;
...
+
+# Check out patch branch from upstream.
+git fetch upstream 9.0.x
+git checkout 9.0.x
+
+# Cherry pick the commit. Use the hash from the commit which was merged
+# into upstream/master, which should be known to your local repo.
+git cherry-pick -x <commit hash from master>
+
+# Resolve merge conflicts if necessary...
```
Comment (kyliau) [inline docs/process/release.md]: or ask PR author to submit a separate commit that targets patch-only

File: docs/process/release.md [hunk 3]
```diff
@@ -39,6 +39,27 @@ Say the following PR is merged;
...
 
+Cherry picking is done by checking out the patch branch and cherry picking the new commit onto it.
+This should be done after merging to master.
+
+```shell
+# Make sure commit to upstream/master is present local repo.
+git fetch upstream master
+
+# Check out patch branch from upstream.
+git fetch upstream 9.0.x
```
Comment (kyliau) [inline docs/process/release.md]: consider making the command more generic, like `git fetch upstream <patch branch>`

File: docs/process/release.md (hunk 2 above)
Comment (dgp1130) [reply docs/process/release.md]: Done.

File: docs/process/release.md (hunk 1 above)
Comment (dgp1130) [reply docs/process/release.md]: Done.

File: docs/process/release.md (hunk 3 above)
Comment (dgp1130) [reply docs/process/release.md]: Done. Added a line at the top to point to what naming convention to look for to identify the patch branch.
